daemon/list.go

```diff
@@ -404,13 +403,12 @@ func portOp(key string, filter map[string]bool) func(value string) error {
 			return fmt.Errorf("filter for '%s' should not contain ':': %s", key, value)
 		}
 		// support two formats, original format <portnum>/[<proto>] or <startport-endport>/[<proto>]
-		proto, portRange := nat.SplitProtoPort(value)
-		start, end, err := nat.ParsePortRange(portRange)
+		portRange, err := containertypes.ParsePortRange(value)
 		if err != nil {
 			return fmt.Errorf("error while looking up for %s %s: %s", key, value, err)
 		}
-		for portNum := start; portNum <= end; portNum++ {
-			filter[fmt.Sprintf("%d/%s", portNum, proto)] = true
+		for portNum := portRange.Start(); portNum <= portRange.End(); portNum++ {
+			filter[fmt.Sprintf("%d/%s", portNum, portRange.Proto())] = true
 		}
```

Comment:
It would be really slick if we provided a convenience method to iterate over the ports of a range.

```suggestion
		for p := range portRange.All() {
			filter[p.String()] = true
}
```

```go
func (pr PortRange) All() iter.Seq[Port] {
	return func(yield func(Port) bool) {
		for i := pr.Start(); i <= pr.End(); i++ {
			if !yield(Port{num: i, proto: pr.proto}) {
				return
			}
		}
	}
}
```